tensorflow_model_optimization/python/core/quantization/keras/prune_preserve/prune_preserve_quantize_registry.py

```diff
@@ -0,0 +1,332 @@
+
+      # SeparableConv need verify from 8bit qat
+      # layers.SeparableConv1D:
+      # _PrunePreserveInfo(['pointwise_kernel'], ['Default8BitConvQuantizeConfig']),
+      # layers.SeparableConv2D:
+      # _PrunePreserveInfo(['pointwise_kernel'], ['Default8BitConvQuantizeConfig']),
+
+      # Embedding need verify from 8bit qat
+      # layers.Embedding: _PrunePreserveInfo(['embeddings'], []),
+
+      layers.ELU: _noPrunePreserve(),
+      layers.LeakyReLU: _noPrunePreserve(),
+      layers.ReLU: _noPrunePreserve(),
+      layers.ThresholdedReLU: _noPrunePreserve(),
+      layers.Softmax: _noPrunePreserve(),
+      layers.Cropping1D: _noPrunePreserve(),
+      layers.Cropping2D: _noPrunePreserve(),
+      layers.Cropping3D: _noPrunePreserve(),
+      layers.UpSampling1D: _noPrunePreserve(),
+      layers.UpSampling2D: _noPrunePreserve(),
+      layers.UpSampling3D: _noPrunePreserve(),
+      layers.ZeroPadding1D: _noPrunePreserve(),
+      layers.ZeroPadding2D: _noPrunePreserve(),
+      layers.ZeroPadding3D: _noPrunePreserve(),
+      layers.Activation: _noPrunePreserve(),
+      layers.ActivityRegularization: _noPrunePreserve(),
+      layers.Dropout: _noPrunePreserve(),
+      layers.Flatten: _noPrunePreserve(),
+      layers.Lambda: _noPrunePreserve(),
+      layers.Masking: _noPrunePreserve(),
+      layers.Permute: _noPrunePreserve(),
+      layers.RepeatVector: _noPrunePreserve(),
+      layers.Reshape: _noPrunePreserve(),
+      layers.SpatialDropout1D: _noPrunePreserve(),
+      layers.SpatialDropout2D: _noPrunePreserve(),
+      layers.SpatialDropout3D: _noPrunePreserve(),
+      layers.Add: _noPrunePreserve(),
+      layers.Average: _noPrunePreserve(),
+      layers.Concatenate: _noPrunePreserve(),
+      layers.Dot: _noPrunePreserve(),
+      layers.Maximum: _noPrunePreserve(),
+      layers.Minimum: _noPrunePreserve(),
+      layers.Multiply: _noPrunePreserve(),
+      layers.Subtract: _noPrunePreserve(),
+      layers.AlphaDropout: _noPrunePreserve(),
+      layers.GaussianDropout: _noPrunePreserve(),
+      layers.GaussianNoise: _noPrunePreserve(),
+      layers.BatchNormalization: _noPrunePreserve(),
+      layers.LayerNormalization: _noPrunePreserve(),
+      layers.AveragePooling1D: _noPrunePreserve(),
+      layers.AveragePooling2D: _noPrunePreserve(),
+      layers.AveragePooling3D: _noPrunePreserve(),
+      layers.GlobalAveragePooling1D: _noPrunePreserve(),
+      layers.GlobalAveragePooling2D: _noPrunePreserve(),
+      layers.GlobalAveragePooling3D: _noPrunePreserve(),
+      layers.GlobalMaxPooling1D: _noPrunePreserve(),
+      layers.GlobalMaxPooling2D: _noPrunePreserve(),
+      layers.GlobalMaxPooling3D: _noPrunePreserve(),
+      layers.MaxPooling1D: _noPrunePreserve(),
+      layers.MaxPooling2D: _noPrunePreserve(),
+      layers.MaxPooling3D: _noPrunePreserve(),
+      TensorFlowOpLayer: _noPrunePreserve(),
+  }
+
+  # TODO(mltools): enable RNN layers, also depend on qat and prune registry
```

Comment:
Depending on the registries directly needs to be discussed since at the moment, those are private APIs.

Reply:
removed